src/main/kotlin/org/rust/lang/core/psi/ext/RsItemElement.kt

```diff
@@ -6,5 +6,10 @@
 package org.rust.lang.core.psi.ext
 
 import org.rust.lang.core.macros.RsExpandedElement
+import org.rust.lang.core.resolve.TYPES
+import org.rust.lang.core.resolve.processNestedScopesUpwards
 
 interface RsItemElement : RsVisibilityOwner, RsOuterAttributeOwner, RsExpandedElement
+
+fun RsItemElement.isInScope(scope: RsElement): Boolean =
```

Comment:
In both case (in type inference and in auto import quick fix) we have the same situation: we have a list of traits and should skip all element that is not in scope. And we call `processNestedScopesUpwards` on each trait. But for all items, we have the same scope so all results of `processNestedScopesUpwards` are the same as well.
Maybe we modify `isInScope` method to call it on list of items, invoke `processNestedScopesUpwards` only one time and just check that elements are in the list? Looks like it's more efficient that is rather important in type inference case